### ISSUE TITLE
Cannot login

### ISSUE THREAD

**Issue by mvl64** [puschie286/fritzbox-bridge#3]
I have a problem logging in to the fritzbox, the app keeps reporting "invalid login".
I have ensured I'm using the correct login details (account and password) and even setup a separate account on the Fritzbox for Homey with a more simple password.
I can login to the fritzbox using both accounts using the web browser.

Note: my homey and fritzbox are in separate network segments, maybe that effects this issue.
I did confirm in the firewall traffic is passed on (port 80 from homey to the fritzbox) - I log all the traffic - fritzbox ip-address is specified correctly in the settings (so, I don't use the default http://fritz.box URL).

Enabling SSL in the app's dialog did not seem to make a difference - same error message is reported back.

**Comment by puschie286** [comment]
hmm sounds like a connection establish problem
-> currently im working on another reported login problem that requires to rewrite the login-process so i will also test for you problem

hope to find some time next week for this ( probably Tuesday or Wednesday )

**Comment by nattlip** [comment]
> I have a problem logging in to the fritzbox, the app keeps reporting "invalid login".
> I have ensured I'm using the correct login details (account and password) and even setup a separate account on the Fritzbox for Homey with a more simple password.
> I can login to the fritzbox using both accounts using the web browser.
> 
> Note: my homey and fritzbox are in separate network segments, maybe that effects this issue.
> I did confirm in the firewall traffic is passed on (port 80 from homey to the fritzbox) - I log all the traffic - fritzbox ip-address is specified correctly in the settings (so, I don't use the default http://fritz.box URL).
> 
> Enabling SSL in the app's dialog did not seem to make a difference - same error message is reported back.

neither can i

**Comment by mvl64** [comment]
> 
> 
> hmm sounds like a connection establish problem
> -> currently im working on another reported login problem that requires to rewrite the login-process so i will also test for you problem
> 
> hope to find some time next week for this ( probably Tuesday or Wednesday )

Thanks ! If you need additional info, please let me know.

**Comment by puschie286** [comment]
sorry for late response 

pushed a update today for better error handling b755d73eca8ed39e7b4e47c5925c0d7a10ae7bb7
but i guess its a different problem.

so you use the same url in settings like for accessing fritzbox via web ?
(i guess you already tested to use http/https infront of your ip)

pls try [Fritzbox IP]/login_sid.lua -> it should show you a xml response

if you still have problems pls provide some informations about your setup ( fritzbox version, fritzbox local ip, homey local ip, fritzbox type ( model ) )

**Comment by mvl64** [comment]
No problem on the delay, we're all busy people and have iimited time
available ... I'm glad do you take the time to review and respond.

I'll do some more testing tonight, but want to already confirm I'm using
the same url as when accessing using the browser.
I tried without and with http:// in the url - not sure whether I tried with
https:// - will reconfirm tonight. I did notice regardless of whether I
tick the 'strict ssl' box the app always connects to port 80 (I was
expecting 443 for ssl).

One question:you mention to try the .lua link - should I enter this in the
app settings, or using a browser accessing the firtzbox ?
And where should Isee the xml ? I'm new to Homey apps and have not yet
found a way to see app logs ...

Talk later

**Comment by puschie286** [comment]
hmm, i can check the port for https -> should be handled by the npm package 'request' but i will check

the [...]login_sid.lua address is for the browser -> your browser should show a xml file
note: there is an issue with something similar that was caused by wrong DNS setup -> [Link](https://github.com/andig/fritzapi/issues/11)

**Comment by mvl64** [comment]
When I access the mentioned URL in the browser - before login, I get the
following response:
<SessionInfo><SID>0000000000000000</SID><Challenge>485e81f6</Challenge><BlockTime>0</BlockTime><Rights/></SessionInfo>

After logging in, it changes to:
<SessionInfo><SID>0000000000000000</SID><Challenge>080edd99</Challenge><BlockTime>0</BlockTime><Rights/></SessionInfo>

Honestly speaking, I'm not clear how this helps but wanted to share anyway
...

I'll have a look at changing HTTPS in the URL to see how that effects the
result.

**Comment by mvl64** [comment]
When I use HTTPS in the URL, the destination port changes to 443 as
expected.
Even when I de-select the "strict SSL" tickbox in the settings, traffic is
pointing to port 443.

I'm not sure how the tickbox effect the traffic - but at least by changing
the URL the port can be controlled.

Unfortunately, neither case results in a proper login :(

Regarding your note on DNS: I'm using IP address in the app settings, so I
don't think DNS would make an impact.

I'll read-up on the link nonetheless and let you know.

**Comment by puschie286** [comment]
Looks good...
It’s pretty sound like a network problem - I guess the fritzbox ip is different in your homey subnet

**Comment by mvl64** [comment]
I've confirmed traffic from Homey is reaching the Fritzbox, by looking at
the firewall logs.
Also on the FritzBox, I see the connection attempt in the event Log - it is
reported there as an incorrect password.
See screenshot below. Note the Homey address is 192.168.30.27.

[image: image.png]

I obviously confirmed the password is correct - it is just one character,
can't go wrong there.

If you still think this is pointing to a network issue, can you please
clarify ?

**Comment by puschie286** [comment]
Sry can’t see the img ( only Text no link )

**Comment by mvl64** [comment]
Ow, sorry - Ill attach it as a file, rather than in place ...
[image: Capture.JPG]

**Comment by puschie286** [comment]
Unfortunately not

**Comment by mvl64** [comment]
Hmm, sorry about that

**Comment by puschie286** [comment]
can you upload the image somewhere ( or use the browser ) ?

currently i have no idea why the fritzbox reject the password/challenge
-> do you have other devices/services that access the fritzbox. trying to login with wrong credentials will cause all active sessions to be destroyed ( found that in the documentation^^ )

and maybe it will help to have the error message, could you post your homey/SimpleLOG log entries when trying to login ?

**Comment by mvl64** [comment]
Hi Puschie,

Sorry, I was out of the country for a couple of days and too busy to follow
up on this.
Tried again today, after I noticed an app update was installed earlier this
week.

Problem still occurs; please refer to file attached.
I have now logged in to the fritzbox with the same account (Homey) on the
web browser, since you mentioned I would be logged out automatically.
That does not occur to me, the browser session is still active whereas you
can see the failed attempt from the Homey device (on 192.168.30.27).
I hope you are able to see this attached file, I've uploaded it instead of
dragging it into the browser.

If not, can you give me an alternative mail address or do you have a way
for me to upload it on github ?

I noticed you mentioned in the change log of this version you've added some
logging.
Is there a way for me to access the logs, and send you some screenshots?

**Comment by puschie286** [comment]
Hey, np but you haven’t post any image link or something. Pls use your browser for github replay to be correct shown

**Comment by mvl64** [comment]
I was replying by email, it seems this removes the attachement (sorry, I never checked this thread on github after creating the initial ticket).
Let me try again now
![Capture](https://user-images.githubusercontent.com/42496758/57974755-a07f7c00-79bd-11e9-9c20-70762b401ba2.JPG)

**Comment by puschie286** [comment]
hmm double checked the password hashing function but everything looks good there ( tested for special keys like € @ ä ö ).
can you share your fritzbox bridge log ? ( install SimpleLOG from app store and restart FritzboxBridge to get all full log with startup and everything )
still think its a subnet problem -> can you try 192.168.30.1 as fritzbox ip ?

**Comment by mvl64** [comment]
My password currently does not use special characters, so that's not it.
Thanks for the tip on SimpleLog - I use PaperTrails myself (to create log entries from flows) and thought SimpleLog did the same.

The error reported is reloaded to "zero depth sigl signed cert" it seems.
See attached file.
Hope this helps !
[Simple LOG.log](https://github.com/puschie286/fritzbox-bridge/files/3197799/Simple.LOG.log)

**Comment by puschie286** [comment]
nice thx :)
can you try with http instead off https and send the log ?
( thanks for note PaperTrails, will add it to supported log apps )

**Comment by mvl64** [comment]
When I try without https, there is no log entry anymore
file is unchanged ...

I've even restarted the the  app to confirm, as you can see in the log:
[Simple LOG2.log](https://github.com/puschie286/fritzbox-bridge/files/3197925/Simple.LOG2.log)

**Comment by puschie286** [comment]
that's bad, sounds like a timeout will happen soon ( 20s - 120s ).

im almost out of idea's how to face this problem :( 

your homey ip is 192.168.**30**.27 and it tries to connect to 192.168.**5**.76
-> are you sure it is reachable ?
i guess because of the different subnet your fritzbox ip is also different for this subnet.
( maybe 192.168.30.76 ? )

**Comment by mvl64** [comment]
Correct, Homey is in the IoT network segment which has no access to internal networks except for some specific devices. I've included a firewall rule to allow Homey to Fritzbox and vice versa - network traffic logging shows there is a flow both ways. 

> -> are you sure it is reachable ?
Yes, I know it is reachable: I can see the login attempt on the FritzBox - and in return it reports "invalid password". 

Both ip addresses are correct, I also see the firewall is logging traffic going from Homey to the Fritzbox.

NOTE: I only use the fritzbox for smart devices now - replaced my router and access points a while ago.
So I could try to move it into the IoT segment too - to see if that changes anything.

NOTE: I cannot change the fritz ip to 192.168.,30.1 - as that is the address of the router.

**Comment by puschie286** [comment]
yes pls try it^^

well, the point is: fritzbox-bridge use fritzboxapi -> fritzboxapi is an implementation of the official fritzbox api. ( also checked the implementation and validate with example data from [documention](https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/Session-ID_english_13Nov18.pdf))

so im not sure how the password can be wrong if you use the correct one and the implementation is correct :/

the last thing i can try is to setup a network with similar configuration and try to reproduce it
( will ask our network specialists for help^^ @work )

**Comment by mvl64** [comment]
Hmm, looking at the code it seems the password is hashed and the hash is sent to validate the password on the box. I'm using a one character password to rule out me making typos, could that be causing issues (is there a minimum length for the MD5 hash?)

**Comment by puschie286** [comment]
Guess not - there is no min length.
  

  
One more idea - can you try to use a laptop and configure it for the smart home subnet and try to login to the fritzbox from there ?
  
  

  
  
>

**Comment by mvl64** [comment]
> so I could try to move it into the IoT segment too - to see if that changes anything.

I have moved the FritzBox into the same network segment as Homey (the IoT network) and it worked right away. Not sure how the API is effected by crossing the router, but apparently it's breaking the login. 
Sorry about this !
I am now able to use the app - will start running some tests.

Thanks a lot for your time and effort.

**Comment by puschie286** [comment]
hehe^^
would be nice if you could share your results here - good luck till then :)

**Comment by mvl64** [comment]
Yes, I confirm: now (after moving FritzBox into the IoT segment) I cannot login anymore from my original network segment (main lan). This issue was unrelated to your app - just unexpected Fritz behaviour !
